// @progress/kendo-angular-grid/dist/npm/filtering/menu/numeric-filter-menu.component.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var core_1 = require("@angular/core");
var utils_1 = require("../../utils");
var kendo_angular_l10n_1 = require("@progress/kendo-angular-l10n");
var numeric_filter_component_1 = require("../numeric-filter.component");
/**
 * Represents a numeric filter menu component.
 *
 * @example
 *  ```ts-no-run
 *      <kendo-grid-column field="UnitPrice" title="Unit Price">
 *          <ng-template kendoGridFilterMenuTemplate let-filter let-column="column" let-filterService="filterService">
 *          <kendo-grid-numeric-filter-menu
 *              [column]="column"
 *              [filter]="filter"
 *              [filterService]="filterService"
 *              >
 *          </kendo-grid-numeric-filter-menu>
 *          </ng-template>
 *      </kendo-grid-column>
 *   ```
 */
var NumericFilterMenuComponent = (function (_super) {
    tslib_1.__extends(NumericFilterMenuComponent, _super);
    function NumericFilterMenuComponent(localization) {
        var _this = _super.call(this, null, localization) || this;
        _this.logicOperators = [
            { text: "And", value: "and" },
            { text: "Or", value: "or" }
        ];
        /**
         * The current menu filter.
         * @type {CompositeFilterDescriptor}
         */
        _this.filter = { filters: [], logic: "and" };
        /**
         * Determines if the second criterion inputs should be shown
         */
        _this.extra = true;
        return _this;
    }
    Object.defineProperty(NumericFilterMenuComponent.prototype, "hostClasses", {
        /**
         * @hidden
         */
        get: function () {
            return false;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericFilterMenuComponent.prototype, "firstFilter", {
        get: function () {
            if (utils_1.isPresent(this.filter) && utils_1.isPresent(this.filter.filters) && this.filter.filters.length) {
                return this.filter.filters[0];
            }
            else {
                return this.insertDefaultFilter(0, {
                    field: (this.column || {}).field,
                    operator: this.currentOperator
                });
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NumericFilterMenuComponent.prototype, "secondFilter", {
        get: function () {
            if (utils_1.isPresent(this.filter) && utils_1.isPresent(this.filter.filters) && this.filter.filters.length > 1) {
                return this.filter.filters[1];
            }
            else {
                return this.insertDefaultFilter(1, {
                    field: (this.column || {}).field,
                    operator: this.currentOperator
                });
            }
        },
        enumerable: true,
        configurable: true
    });
    NumericFilterMenuComponent.prototype.logicChange = function (value) {
        this.filter.logic = value;
    };
    NumericFilterMenuComponent.prototype.insertDefaultFilter = function (index, filter) {
        this.filter = (this.filter || { filters: [], logic: "and" });
        this.filter.filters[index] = filter;
        return filter;
    };
    return NumericFilterMenuComponent;
}(numeric_filter_component_1.NumericFilterComponent));
NumericFilterMenuComponent.decorators = [
    { type: core_1.Component, args: [{
                selector: 'kendo-grid-numeric-filter-menu',
                template: "\n        <kendo-grid-numeric-filter-menu-input\n            [currentFilter]=\"firstFilter\"\n            [operators]=\"operators\"\n            [filterService]=\"filterService\"\n            [column]=\"column\"\n            [filter]=\"filter\"\n            [format]=\"format\"\n            [decimals]=\"decimals\"\n            [spinners]=\"spinners\"\n            [min]=\"min\"\n            [max]=\"max\"\n            [step]=\"step\"\n            >\n        </kendo-grid-numeric-filter-menu-input>\n        <kendo-dropdownlist\n            *ngIf=\"extra\"\n            class=\"k-filter-and\"\n            [data]=\"logicOperators\"\n            [valuePrimitive]=\"true\"\n            (valueChange)=\"logicChange($event)\"\n            [value]=\"filter?.logic\"\n            textField=\"text\"\n            valueField=\"value\">\n        </kendo-dropdownlist>\n        <kendo-grid-numeric-filter-menu-input\n            *ngIf=\"extra\"\n            [operators]=\"operators\"\n            [currentFilter]=\"secondFilter\"\n            [filterService]=\"filterService\"\n            [column]=\"column\"\n            [filter]=\"filter\"\n            [format]=\"format\"\n            [decimals]=\"decimals\"\n            [spinners]=\"spinners\"\n            [min]=\"min\"\n            [max]=\"max\"\n            [step]=\"step\"\n            >\n        </kendo-grid-numeric-filter-menu-input>\n    "
            },] },
];
/** @nocollapse */
NumericFilterMenuComponent.ctorParameters = function () { return [
    { type: kendo_angular_l10n_1.LocalizationService, },
]; };
NumericFilterMenuComponent.propDecorators = {
    'hostClasses': [{ type: core_1.HostBinding, args: ['class.k-filtercell',] },],
    'column': [{ type: core_1.Input },],
    'filter': [{ type: core_1.Input },],
    'extra': [{ type: core_1.Input },],
    'filterService': [{ type: core_1.Input },],
};
exports.NumericFilterMenuComponent = NumericFilterMenuComponent;
